public/js/clients/firefox.js

```diff
@@ -9,6 +9,58 @@ let currentClient = null;
 let currentThreadClient = null;
 let currentTabTarget = null;
 
+// API implementation
+
+let APIClient = {
+  resume() {
+    return new Promise(resolve => {
+      currentThreadClient.resume(resolve);
+    });
+  },
+
+  stepIn() {
+    return new Promise(resolve => {
+      currentThreadClient.stepIn(resolve);
+    });
+  },
+
+  stepOver() {
+    return new Promise(resolve => {
+      currentThreadClient.stepOver(resolve);
+    });
+  },
+
+  stepOut() {
+    return new Promise(resolve => {
+      currentThreadClient.stepOut(resolve);
+    });
+  },
+
+  getSources() {
+    return currentThreadClient.getSources();
+  },
+
+  sourceContents(sourceId) {
+    const sourceClient = currentThreadClient.source({ actor: sourceId });
+    return sourceClient.source();
+  },
+
+  setBreakpoint(location, condition) {
+    const sourceClient = currentThreadClient.source({ actor: location.sourceId });
+    return sourceClient.setBreakpoint({
+      line: location.line,
+      column: location.column,
+      condition: condition
+    });
+  }
+};
+
+function getAPIClient() {
+  return APIClient;
+}
```

Comment:
I agree, let's do that as a follow-up. I started feeling the same way as I was writing this.